dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/bundle/TrackerBundle.java

```diff
@@ -199,4 +192,19 @@ public String getUsername()
             .filter( e -> getEvent( e.getTrackerObjectUid() ).isPresent() )
             .collect( Collectors.toMap( RuleEffects::getTrackerObjectUid, RuleEffects::getRuleEffects ) );
     }
+
+    public TrackerImportStrategy setStrategy( Event event, TrackerImportStrategy strategy )
+    {
+        return this.getResolvedStrategyMap().get( Event.class ).put( event.getUid(), strategy );
+    }
+
+    public TrackerImportStrategy setStrategy( Enrollment enrollment, TrackerImportStrategy strategy )
+    {
+        return this.getResolvedStrategyMap().get( Enrollment.class ).put( enrollment.getUid(), strategy );
+    }
```

Comment:
Couldn't this be only one method using generics? Haven't tried but something like

`    public <T extends TrackerDto> TrackerImportStrategy setStrategy( T trackerDto, Trackertype trackerType, TrackerImportStrategy strategy ) {
....
}
`

Reply:
Yes, I just changed so many things in this PR and this is used in a lot of places.
I rather leave that for another time